palantir-java-format/src/main/java/com/palantir/javaformat/java/java14/Java14InputAstVisitor.java

```diff
@@ -0,0 +1,237 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.javaformat.java.java14;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.collect.MoreCollectors.toOptional;
+import static com.sun.source.tree.Tree.Kind.BLOCK;
+
+import com.google.common.base.Verify;
+import com.google.common.collect.ImmutableList;
+import com.palantir.javaformat.Op;
+import com.palantir.javaformat.OpsBuilder;
+import com.palantir.javaformat.java.JavaInputAstVisitor;
+import com.sun.source.tree.BindingPatternTree;
+import com.sun.source.tree.CaseTree;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.InstanceOfTree;
+import com.sun.source.tree.SwitchExpressionTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.tree.YieldTree;
+import com.sun.tools.javac.code.Flags;
+import com.sun.tools.javac.tree.JCTree;
+import com.sun.tools.javac.tree.TreeInfo;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Extends {@link JavaInputAstVisitor} with support for AST nodes that were added or modified for Java 14.
+ */
+public class Java14InputAstVisitor extends JavaInputAstVisitor {
```

Comment:
I'm wondering what the longer-term implications of this pattern are - e.g. java 15 introduces a teeny bit more syntax right (specifically the `sealed` keyword I think)... would the idea be to have one of these classes for each of the java versions we support compiling with? and then at some point take a hard stance and increment our required min version, collapsing some of these classes into one?